lib/Resource.js

```diff
@@ -149,12 +162,12 @@ Resource.prototype._addRelation = function(relationName, rawRelation, resource)
 };
 Resource.prototype._removeRelation = function(relationName, rawRelation, resource) {
   this._checkIsResource(resource);
-  if (rawRelation.data instanceof Array) {
+  if (Resource._rawRelationHasMany(rawRelation)) {
     rawRelation.data = rawRelation.data.filter(function(id) {
-      return !resource._matchesId(id);
+      return !resource._matches(id);
     });
     this[relationName] = this[relationName].filter(function(id) {
-      return !resource._matchesId(id);
+      return !resource._matches(id);
```

Comment:
Yes. It was a typo. It might not ever have worked. The new system test failed on it and that's why I spotted it.